src/data/calendar.js

```diff
@@ -0,0 +1,136 @@
+        var globalIndex = weekIndex * 7 + dayIndex;
+        var dayOfMonth = globalIndex - firstWeekDayOfMonth + 1;
+        day.date = new Date(year, month_number-1, dayOfMonth);
+
+        if (day.date.getMonth() === month_number - 1) {
+          var currentDate = year + "-" + zfill(day.date.getMonth() + 1) + "-" + zfill(dayOfMonth);
+          day.hours = 8;
+          if ([0, 6].includes(day.date.getDay())) {
+            day.hours = 8;
+            day.type = 'WE';
+          } else if (bankHolidays.includes(currentDate)) {
+            day.type = 'JF';
+          } else if (boxingDays.includes(currentDate)) {
+            day.type = 'CS';
+          } else {
+            day.type = 'JT';
+          }
+        } else {
+          day.type = null;
+        }
+        return day;
+      });
+    });
+  }
+
+  function updateWeeksWithHolidays(state) {
+    // Update the weeks data with the PTO infos for the current month.
+    state.holidays.forEach((holiday) => {
+      // Is in current month?
+      var start = new Date(holiday.start);
+      var end = new Date(holiday.end);
+
+      // Handle case where holidays ends on a week-end
+      while ([0, 6].includes(end.getDay())) {
+        end.setDate(end.getDate() - 1);
+      }
+
+      if (start.getMonth() <= state.currentMonth - 1 && start.getFullYear() <= state.currentYear &&
+          end.getMonth() >= state.currentMonth -1 && end.getFullYear() >= state.currentYear) {
+        // Someday of this holiday are in the current month
+        var hours = holiday.hours;
+        var type = guessTypeFromComment(holiday.comment);
+
+        state.weeks.forEach(week => {
+          week.forEach(day => {
+            var currentDayText = day.date.getFullYear() + '-' + zfill(day.date.getMonth() - 1) + '-' + zfill(day.date.getDate());
+            var startDayText = start.getFullYear() + '-' + zfill(start.getMonth() - 1) + '-' + zfill(start.getDate());
+            var endDayText = end.getFullYear() + '-' + zfill(end.getMonth() - 1) + '-' + zfill(end.getDate());
+
+            if (startDayText <= currentDayText && endDayText >= currentDayText) {
+              var isLast = endDayText == currentDayText;
+              if (day.type !== 'WE') {
+                if (day.type == 'JT') {
+                  day.type = type;
+                  // If there is still more than 8 hours, it is probably a full day off
+                  if (hours > 8) {
+                    hours -= 8;
+                  } else if (hours > 0) {
+                    if (!isLast) {
+                      day.hours = 4;
+                      hours -= 4;
+                    } else {
+                      // If there is less, it is probably an half day off.
+                      day.hours = hours;
+                      hours = 0;
+                    }
+                  }
+                } else {
+                  // A CS or JF is always 8 hours
+                  hours -= 8;
+                }
+              }
```

Comment:
Mmm I don't think we should remove the hours for JF, actually :) try to add a PTO from pto.mozilla.org over a JF, you'll see there is no counted hours for the JF.